[CI] EsqlRetryIT testQueryWhileDeletingIndices failing

**Build Scans:**
- [elasticsearch-periodic #11954 / adoptopenjdk17_checkpart3_java-fips-matrix](https://gradle-enterprise.elastic.co/s/pvgcaob65d2s4)
- [elasticsearch-periodic #11954 / adoptopenjdk17_checkpart3_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/i5d6ztepgfpvq)
- [elasticsearch-periodic #11954 / adoptopenjdk17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/mhgnnhq7g2bvg)
- [elasticsearch-periodic #11954 / graalvm-ce17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/bntgnlkfhzyyq)
- [elasticsearch-pull-request #125525 / part-3](https://gradle-enterprise.elastic.co/s/k27pvyh2fh5f4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.EsqlRetryIT.testQueryWhileDeletingIndices" -Dtests.seed=833F55E0A92AE8E8 -Dtests.locale=sl -Dtests.timezone=America/Cuiaba -Druntime.java=17
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.EsqlRetryIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testQueryWhileDeletingIndices'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Leftover exchanges ExchangeService{sinks=[VrSc_DsPS7WHeCk1x8uLxg:150/2, VrSc_DsPS7WHeCk1x8uLxg:150/2[n]]} on node node_s0
```

**Issue Reasons:**
- [8.19] 5 failures in test testQueryWhileDeletingIndices (9.3% fail rate in 54 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -342,12 +342,6 @@ tests:
 - class: org.elasticsearch.readiness.ReadinessClusterIT
   method: testReadinessDuringRestartsNormalOrder
   issue: https://github.com/elastic/elasticsearch/issues/142440
-- class: org.elasticsearch.xpack.esql.action.EsqlRetryIT
-  method: testQueryWhileDeletingIndices
-  issue: https://github.com/elastic/elasticsearch/issues/143639
-- class: org.elasticsearch.xpack.esql.action.EsqlRetryIT
-  method: testRetryOnShardFailures
-  issue: https://github.com/elastic/elasticsearch/issues/143640
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   method: test {stats.StatsByMvFieldAggExpr}
   issue: https://github.com/elastic/elasticsearch/issues/141835

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeComputeHandler.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeComputeHandler.java
@@ -335,8 +335,8 @@ final class DataNodeComputeHandler implements TransportRequestHandler<DataNodeRe
             }
             final var doAcquire = ActionRunnable.supply(listener, () -> {
                 final List<SearchContext> searchContexts = new ArrayList<>(targetShards.size());
-                SearchContext context = null;
                 for (IndexShard shard : targetShards) {
+                    SearchContext context = null;
                     try {
                         var aliasFilter = aliasFilters.getOrDefault(shard.shardId().getIndex(), AliasFilter.EMPTY);
                         var shardRequest = new ShardSearchRequest(

